File: exercise/php/day23/src/App/Main.php
<?php

namespace App;

use Core\Control\ControlSystem;
use Core\Control\SleighNotStartedException;
use Core\Control\ReindeersNeedRestException;

class Main
{
    public static function main()
    {
        $controlSystem = new ControlSystem();
        $controlSystem->startSystem();

        $handle = fopen("php://stdin", "r");
        $keepRunning = true;

        while ($keepRunning) {
            echo "Enter a command (ascend (a), descend (d), park (p), or quit (q)): ";
            $command = trim(fgets($handle));

            switch ($command) {
                case 'ascend':
                case 'a':
                    try {
                        $controlSystem->ascend();
                    } catch (ReindeersNeedRestException | SleighNotStartedException $e) {
                        echo $e->getMessage() . PHP_EOL;
                    }
                    break;
                case 'descend':
                case 'd':
                    try {
                        $controlSystem->descend();
                    } catch (SleighNotStartedException $e) {
                        echo $e->getMessage() . PHP_EOL;
                    }
                    break;
                case 'park':
                case 'p':
                    try {
                        $controlSystem->park();
                    } catch (SleighNotStartedException $e) {
                        echo $e->getMessage() . PHP_EOL;
                    }
                    break;
                case 'quit':
                case 'q':
                    $keepRunning = false;
                    break;
                default:
                    echo "Invalid command. Please try again." . PHP_EOL;
            }
        }

        fclose($handle);
        $controlSystem->stopSystem();
    }
}

Main::main();

File: exercise/scala/day23/src/main/scala/app/Main.scala
package app

import core.control.{ControlSystem, ReindeersNeedRestException, SleighNotStartedException}

import scala.io.StdIn

object Main {
  def main(args: Array[String]): Unit = {
    val controlSystem = new ControlSystem()
    controlSystem.startSystem()

    var keepRunning = true

    while (keepRunning) {
      println("Enter a command (ascend (a), descend (d), park (p), or quit (q)): ")
      val command = StdIn.readLine()

      command match {
        case "ascend" | "a" =>
          try {
            controlSystem.ascend()
          } catch {
            case e: ReindeersNeedRestException => println(e.getMessage)
            case e: SleighNotStartedException => println(e.getMessage)
          }
        case "descend" | "d" =>
          try {
            controlSystem.descend()
          } catch {
            case e: SleighNotStartedException => println(e.getMessage)
          }
        case "park" | "p" =>
          try {
            controlSystem.park()
          } catch {
            case e: SleighNotStartedException => println(e.getMessage)
          }
        case "quit" | "q" =>
          keepRunning = false
        case _ =>
          println("Invalid command. Please try again.")
      }
    }

    controlSystem.stopSystem()
  }
}
